dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/LinkageChecker.java

```diff
@@ -186,39 +196,41 @@ private LinkageChecker(
 
     // Filter classes in whitelist
     SetMultimap<SymbolProblem, ClassFile> filteredMap =
-        Multimaps.filterEntries(problemToClass.build(), LinkageChecker::problemFilter);
+        Multimaps.filterEntries(problemToClass.build(), this::problemFilter);
     return ImmutableSetMultimap.copyOf(filteredMap);
   }
 
+  private static ImmutableList<LinkageErrorMatcher> readExclusionMatchers() throws IOException {
+    ImmutableList.Builder<LinkageErrorMatcher> exclusionMatchers = ImmutableList.builder();
+
+    try {
+      URL defaultRuleUrl =
+          LinkageChecker.class
+              .getClassLoader()
+              .getResource("linkage-checker-exclusion-default.xml");
+      ImmutableList<LinkageErrorMatcher> defaultMatchers =
+          ExclusionFileParser.parse(defaultRuleUrl);
+      exclusionMatchers.addAll(defaultMatchers);
+    } catch (SAXException | VerifierConfigurationException ex) {
+      throw new IOException("Could not read default exclusion rule", ex);
+    }
+
+    return exclusionMatchers.build();
+  }
+
   /**
    * Returns true if the linkage error {@code entry} should be reported. False if it should be
    * suppressed.
    */
-  private static boolean problemFilter(Map.Entry<SymbolProblem, ClassFile> entry) {
-    ClassFile classFile = entry.getValue();
+  private boolean problemFilter(Map.Entry<SymbolProblem, ClassFile> entry) {
+    ClassFile sourceClass = entry.getValue();
     SymbolProblem symbolProblem = entry.getKey();
-    String sourceClassName = classFile.getBinaryName();
-    if (SOURCE_CLASSES_TO_SUPPRESS.contains(sourceClassName)) {
-      return false;
-    }
-
-    // GraalVM-related libraries depend on Java Compiler Interface (JVMCI) that only exists in
-    // special JDK. https://github.com/GoogleCloudPlatform/cloud-opensource-java/issues/929
-    String problematicClassName = symbolProblem.getSymbol().getClassBinaryName();
-    if (problematicClassName.startsWith("jdk.vm.ci")
-        && (sourceClassName.startsWith("com.oracle.svm")
-            || sourceClassName.startsWith("com.oracle.graal")
-            || sourceClassName.startsWith("org.graalvm"))) {
-      return false;
-    }
 
-    // Mockito's MockMethodDispatcher uses special class loader to load MockMethodDispatcher.raw
-    // https://github.com/GoogleCloudPlatform/cloud-opensource-java/issues/407
-    if (problematicClassName.equals("org.mockito.internal.creation.bytebuddy.MockMethodDispatcher")
-        && sourceClassName.startsWith("org.mockito.internal.creation.bytebuddy")) {
-      return false;
+    for (LinkageErrorMatcher matcher : exclusionMatchers) {
```

Comment:
This list iteration logic could become part of an ExcludedErrors class. Then it becomes simply 

return excludedErrors.contains(symbolProblem, sourceClass)